R/simulateGLV.R

```diff
@@ -75,24 +77,31 @@ dxdt <- function(t, x, parameters){
 
 setMethod("simulateGLV", signature = c(n.species="numeric"),
     function(n.species, A, x = runif(n.species),
-                b = runif(n.species), t.start = 0, t.store, norm = FALSE, ...){
+            b = runif(n.species), t.start = 0, t.store, norm = FALSE,
+            arguments = TRUE, ...){
         parameters <- cbind(b, A)
-        t.dyn <- simulateTimeSeries(t.start, ..., t.store)
+        t.dyn <- SimulationTimes(t.start, ..., t.store)
 
         out <- ode(
                 y = x,
                 times = t.dyn$t.sys,
                 func = dxdt,
                 parms = parameters
-        )
+    )
         spab <- t(out[,2:ncol(out)])
         spab <- spab[,t.dyn$t.index]
         print(str(spab))
+
         if(norm){
         spab <- t(t(spab)/colSums(spab))
-            }
+    }
         spab
         SE <- SummarizedExperiment(assays = list(counts=spab))
+        if (arguments) {
+            metadata(SE) <- list(x  = x,
+                                A = A,
+                                b = b)
+    }
```

Comment:
@danielriosgarza  This part adds the parameters into SummarizedExperiment object as a field of metadata. I tried to do a general function in order to apply to all functions, but then all parameters including n.species., t.start etc. are also added as a field. If you agree to this version that I added to simulateGLV, I can apply this to all other functions.

Reply:
Having all the parameters is actually better. Also, this should be the general behavior so no need for the conditional and the parameter arguments.